lib/google_maps/response.ex

```diff
@@ -7,11 +7,15 @@ defmodule GoogleMaps.Response do
 
   @type error :: HTTPoison.Error.t | status()
 
-  def wrap({:error, error}), do: {:error, error}
-  def wrap({:ok, %{body: body} = response}) when is_binary(body) do
-    wrap({:ok, %{response | body: Jason.decode!(body)}})
+  def wrap({:error, error}, _resource), do: {:error, error}
```

Comment:
Is it possible to look at the response header to determine that the response is binary instead of an additional parameter to the function? There should be a `headers` key in `response` map.

Reply:
Yup, definitely possible! Though the `headers` key spits out a keyword list with the default HTTPoison requester, so I believe we cannot pattern match our way out of this one entirely. 

What do you think of rearranging so that we grab the content type in `wrap/1` and then put type and result-specific logic in a private function? e.g.

```elixir
def wrap({:error, error}), do: {:error, error}
def wrap({:ok, %{headers: headers} = result}) do
  content_type = Keyword.get(headers, "Content-Type")
  do_wrap(result, content_type)
end

defp do_wrap(%{body: body} = response}, "application/json") when is_binary(body) do
  do_wrap({:ok, %{response | body: Jason.decode!(body)}})
end
# etc
```

Reply:
Would it be any better to modify `Request` to return headers as a map?